test/integration/service-api/basic_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2022 Authors of Global Resource Service.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service_api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"sync"
+	"testing"
+
+	"global-resource-service/resource-management/pkg/common-lib/types"
+	"global-resource-service/resource-management/pkg/common-lib/types/event"
+	"global-resource-service/resource-management/pkg/distributor"
+	"global-resource-service/resource-management/pkg/distributor/storage"
+	"global-resource-service/resource-management/pkg/service-api/endpoints"
+	apitypes "global-resource-service/resource-management/pkg/service-api/types"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/klog"
+)
+
+//TODO: will have mock interface impl once the interface is moved out to the common lib
+var existedNodeId = make(map[uuid.UUID]bool)
+var rvToGenerate = 0
+
+var singleTestLock = sync.Mutex{}
+
+func setUp() *distributor.ResourceDistributor {
+	singleTestLock.Lock()
+	return distributor.GetResourceDistributor()
+}
+
+func tearDown(resourceDistributor *distributor.ResourceDistributor) {
+	defer singleTestLock.Unlock()
+}
+
+func createRandomNode(rv int) *types.LogicalNode {
+	id := uuid.New()
+
+	return &types.LogicalNode{
+		Id:              id.String(),
+		ResourceVersion: strconv.Itoa(rv),
+		GeoInfo: types.NodeGeoInfo{
+			Region:            0,
+			ResourcePartition: 0,
+		},
+	}
+}
+
+func generateAddNodeEvent(eventNum int) []*event.NodeEvent {
+	result := make([]*event.NodeEvent, eventNum)
+	for i := 0; i < eventNum; i++ {
+		rvToGenerate += 1
+		node := createRandomNode(rvToGenerate)
+		nodeEvent := event.NewNodeEvent(node, event.Added)
+		result[i] = nodeEvent
+	}
+	return result
+}
+
+func TestHttpGet(t *testing.T) {
```

Comment:
I am confused. What is the difference between this test and TestHttpGet in installer_test.go?